Updated build-contrib.md to make it an actually decent setup guide

DatComputerNerd opened the pull request:
Completely rewrote some sections to add clarity, specifically for beginners. If i missed anything please let me know

DatComputerNerd committed:
Updated build-contrib.md to make it an actually decent setup guide

DexrnZacAttack commented:
Are you able to mention the UI proj in here too? since it's much more user friendly

DatComputerNerd commented:
Yeah sure. I'll update it sometime later.

DatComputerNerd committed:
Update build-contrib.md to mention the UI project

Added in a separate section on installing the game through the UI. Also added a section talking about managing saves

DexrnZacAttack commented:
Update the guide to mention using the UI proj for both registering and patching the packages

DexrnZacAttack commented:
Github mobile won't let me add a review comment
